interfaces/repo.go

```diff
@@ -679,6 +607,26 @@ func (r *Repository) Connect(ref *ConnRef, plugStaticAttrs, plugDynamicAttrs, sl
 	return conn, nil
 }
 
+// NotConnectedError is returned by Disconnect() if the requested connection does
+// not exist.
+type NotConnectedError struct {
+	message string
+}
+
+func (e *NotConnectedError) Error() string {
+	return e.message
+}
+
+// NoPlugOrSlotError is returned by Disconnect() if either the plug or slot does
+// no exist.
+type NoPlugOrSlotError struct {
```

Comment:
This is currently unused beyond returning. Do you have plans for it?

Reply:
Yes, in the followup. FWTW I've just discovered we had very similiar UnknownPlugOrSlotError before in this package, I've removed the old one in favor of this.